README.rst

```diff
@@ -84,7 +99,7 @@ by each aggregator, and values are summed recursively from the course block on
 down.  Values for every node in the whole tree can be calculated in a single
 traversal.  These calculations can either be performed "read-only" (to get the
 latest data for each user), or "read-write" to store that data in the
-[`completion_aggregator.Aggregator` model](https://github.com/open-craft/openedx-completion-aggregator/blob/master/completion_aggregator/models.py#L199).
+`completion_aggregator.Aggregator model <https://github.com/open-craft/openedx-completion-aggregator/blob/a71ab4f077/completion_aggregator/models.py#L199>`_.
 
 During regular course interaction, a learner will calculate aggregations on the
 fly to get the latest information.  However, on-the-fly calculations are too
```

Comment:
The documentation right after this mentions, `every hour or less` but this is not enforced by the code. Is this our recommendation? Should we have a recommendation?

Reply:
I just realized that we forgot to update docs in #67. This introduced locking mechanisms for batch aggregation and locking operations. Without this there was a possibility to creating an endless loop when aggregator ran too often - if it wouldn't be able to complete the whole operation in one cycle, it would start adding new task messages to Redis and they queue would be growing infinitely, as the tasks from the ongoing run would be duplicated.
So we added locks to prevent simultaneous runs, but if one run can take more than 30 minutes, the `COMPLETION_AGGREGATOR_AGGREGATION_LOCK_TIMEOUT_SECONDS` variable should be increased to prevent releasing the lock too early as a result of a timeout. Unfortunately the max duration of a single run will depend on the platform's usage, so it's not possible to provide sane values for everyone here.

Reply:
I guess it would ideally just be a process that constantly runs again as soon as it's finished, unless there are no stale completions. But yeah, I think it really depends on the usage of the instance. How about if I just say "e.g. hourly"

For now I've just said

> (e.g. every hour, but this is configurable)

@Agrendalath Thanks for that information - I've added this:

> 3. A lock is used to prevent overlapping runs of ``run_aggregator_service``. If you have a lot of data on your instance such that a single run of ``run_aggregator_service`` takes longer than 30 minutes, you must configure ``COMPLETION_AGGREGATOR_AGGREGATION_LOCK_TIMEOUT_SECONDS`` to a larger value to prevent releasing the lock too early as a result of a timeout.